pkg/client/kubernetes/cache/cache_test.go

```diff
@@ -0,0 +1,237 @@
+		})
+
+		It("should return a cache error for non API errors", func() {
+			shoot := &gardencorev1beta1.Shoot{
+				ObjectMeta: objectMeta,
+			}
+			fallback.EXPECT().Get(ctx, client.ObjectKeyFromObject(shoot), shoot).Return(fmt.Errorf("foo"))
+			err := aggregator.Get(ctx, client.ObjectKeyFromObject(shoot), shoot)
+			Expect(err).To(BeCacheError())
+		})
+	})
+
+	Describe("#List", func() {
+		It("should list Secrets from special cache", func() {
+			secrets := &corev1.SecretList{}
+			secretCache.EXPECT().List(ctx, secrets).Return(nil)
+			Expect(aggregator.List(ctx, secrets)).To(Succeed())
+		})
+
+		It("should list ConfigMaps from fallback cache", func() {
+			cms := &corev1.ConfigMapList{}
+			fallback.EXPECT().List(ctx, cms).Return(nil)
+			Expect(aggregator.List(ctx, cms)).To(Succeed())
+		})
+
+		It("should list Shoots from fallback cache", func() {
+			shoots := &gardencorev1beta1.ShootList{}
+			fallback.EXPECT().List(ctx, shoots).Return(nil)
+			Expect(aggregator.List(ctx, shoots)).To(Succeed())
+		})
+	})
+
+	Describe("#GetInformer", func() {
+		It("should get informer for Secret from special cache", func() {
+			secret := &corev1.Secret{}
+			secretCache.EXPECT().GetInformer(ctx, secret)
+			Expect(aggregator.GetInformer(ctx, secret)).To(Succeed())
+		})
+
+		It("should get informer for ConfigMap from fallback cache", func() {
+			cm := &corev1.ConfigMap{}
+			fallback.EXPECT().GetInformer(ctx, cm)
+			Expect(aggregator.GetInformer(ctx, cm)).To(Succeed())
+		})
+	})
+
+	Describe("#GetInformerForKind", func() {
+		It("should get informer for Secret from special cache", func() {
+			gvk := corev1.SchemeGroupVersion.WithKind("Secret")
+			secretCache.EXPECT().GetInformerForKind(ctx, gvk)
+			Expect(aggregator.GetInformerForKind(ctx, gvk)).To(Succeed())
+		})
+
+		It("should get informer for SecretList from special cache", func() {
+			gvk := corev1.SchemeGroupVersion.WithKind("SecretList")
+			secretCache.EXPECT().GetInformerForKind(ctx, gvk)
+			Expect(aggregator.GetInformerForKind(ctx, gvk)).To(Succeed())
+		})
+
+		It("should get informer for ConfigMap from fallback cache", func() {
+			gvk := corev1.SchemeGroupVersion.WithKind("ConfigMap")
+			fallback.EXPECT().GetInformerForKind(ctx, gvk)
+			Expect(aggregator.GetInformerForKind(ctx, gvk)).To(Succeed())
+		})
+	})
+
+	Describe("#Start", func() {
+		It("should start all informers", func() {
+			informerCtx, cancel := context.WithCancel(ctx)
+			testChan := make(chan struct{})
+
+			fallback.EXPECT().Start(informerCtx).DoAndReturn(func(ctx context.Context) error {
+				testChan <- struct{}{}
+				return nil
+			})
+			secretCache.EXPECT().Start(informerCtx).DoAndReturn(func(ctx context.Context) error {
+				testChan <- struct{}{}
+				return nil
+			})
+
+			go func() {
+				for count := 0; count < 2; count++ {
+					<-testChan
+				}
+				cancel()
+			}()
+
+			Expect(aggregator.Start(informerCtx)).To(Succeed())
```

Comment:
Hmm, I think this test would not terminate, if some errors occur under the hood and `Start` is not called on either of the caches. Can you improve the test to make use of ginkgo's build-in [async patterns](https://onsi.github.io/ginkgo/#asynchronous-tests)?
E.g. like this (applies a default timeout of `1s` by adding a `Done` parameter)
```suggestion
			go func() {
				defer GinkgoRecover()
				Expect(aggregator.Start(informerCtx)).To(Succeed())
			}()

			Eventually(testChan).Should(Receive())
			Eventually(testChan).Should(Receive())
			cancel()
			close(done)
```